src/proxy/mod.rs

```diff
@@ -0,0 +1,200 @@
+use crate::ctx::{AppConfig, GenericResult, ProxyRoute};
+use crate::sign::SignedMessage;
+use hyper::header;
+use hyper::header::{HeaderName, HeaderValue};
+use hyper::{Body, Request, Response, StatusCode, Uri};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
+mod moralis;
+use moralis::{proxy_moralis, validation_middleware_moralis};
+mod quicknode;
+pub(crate) use quicknode::{
+    proxy_quicknode, validation_middleware_quicknode, QuicknodePayload, QuicknodeSocketPayload,
+};
+
+const X_AUTH_PAYLOAD: &str = "X-Auth-Payload";
+const KEEP_ALIVE: &str = "keep-alive";
+
+/// Enumerates different proxy types supported by the application, focusing on separating feature logic.
+/// This allows for differentiated handling based on what the proxy should do with the request,
+/// directing each to the appropriate service or API based on its designated proxy type.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub(crate) enum ProxyType {
+    Quicknode,
+    Moralis,
+}
+
+/// Represents the types of payloads that can be processed by the proxy, with each variant tailored to a specific proxy type.
+/// This helps in managing the logic for routing and processing requests appropriately within the proxy layer.
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) enum PayloadData {
+    Quicknode {
+        payload: QuicknodePayload,
+        signed_message: SignedMessage,
+    },
+    /// Moralis feature requires only Signed Message in X-Auth-Payload header
+    Moralis(SignedMessage),
```

Comment:
Since https://github.com/KomodoPlatform/komodo-defi-proxy/pull/23 this is same for quicknode too if I am not mistaken.

Reply:
It means that Moralis feature in Proxy layer needs only sig message from us, while Quicknode needs sig message + payload. I will add doc comm for Quicknode and update Moralis doc to avoid confusion